mate-panel/button-widget.c

```diff
@@ -370,10 +387,21 @@ button_widget_draw (GtkWidget *widget,
 		(state_flags & GTK_STATE_FLAG_PRELIGHT) && (state_flags & GTK_STATE_FLAG_ACTIVE)) ?
 		BUTTON_WIDGET_DISPLACEMENT * height / 48.0 : 0;
 
-	w = cairo_image_surface_get_width (button_widget->priv->surface) / scale;
-	h = cairo_image_surface_get_height (button_widget->priv->surface) / scale;
-	x = off + (width - w) / 2;
-	y = off + (height - h) / 2;
+	display = gdk_display_get_default ();
+	if  (GDK_IS_X11_DISPLAY (display))
+	{
+		w = cairo_image_surface_get_width (button_widget->priv->surface) / scale;
+		h = cairo_image_surface_get_height (button_widget->priv->surface) / scale;
+		x = off + (width - w) / 2;
```

Comment:
Can this be placed outside of conditional judgment? This can simplify the code

Reply:
No it cannot be: x11 and Wayland treat this differently, with wayland scaling by default and x11 requiring explicit scaling. Otherwise you get buttons offset and cropped in either Wayland or x11, depending on which code block you enable unconditionally.

Reply:
What I mean is that the values of x and y can be placed outside the conditional judgment
```
x = off + (width - w) / 2;
y = off + (height - h) / 2;
```
Of course, it doesn't have much impact, so I agree with this PR